examples/post_training_quantization/torch/ssd300_vgg16/main.py

```diff
@@ -0,0 +1,186 @@
+        super().__init__()
+        self.transform = transform
+        self.data_path = Path(data_path)
+        self.images_path = self.data_path / 'images' / 'train2017'
+        self.labels_path = self.data_path / 'labels' / 'train2017'
+        self.image_ids = sorted(map(lambda p: int(p.stem), self.images_path.glob("*.jpg")))
+
+    def __getitem__(self, item):
+        image_id = self.image_ids[item]
+
+        img = Image.open(self.images_path / f'{image_id:012d}.jpg')
+        img_w, img_h = img.size
+        target = dict(image_id=[image_id], boxes=[], labels=[])
+        label_filepath = self.labels_path / f'{image_id:012d}.txt'
+        if label_filepath.exists():
+            with open(label_filepath, 'r') as f:
+                for box_descr in f.readlines():
+                    category_id, rel_x, rel_y, rel_w, rel_h = tuple(map(float, box_descr.split(' ')))
+                    box_x1, box_y1 = img_w * (rel_x - rel_w / 2), img_h * (rel_y - rel_h / 2)
+                    box_x2, box_y2 = img_w * (rel_x + rel_w / 2), img_h * (rel_y + rel_h / 2)
+                    target["boxes"].append((box_x1, box_y1, box_x2, box_y2))
+                    target["labels"].append(self.category_mapping[int(category_id)])
+
+        for k in target.keys():
+            target[k] = torch.as_tensor(target[k], dtype=torch.float32 if k == "boxes" else torch.int64)
+        img, target = self.transform(img, target)
+        return img, target
+
+    def __len__(self):
+        return len(self.image_ids)
+
+
+def validate(model, dataset, device):
+    model.to(device)
+    model.eval()
+    metric = MeanAveragePrecision()
+    with torch.no_grad():
+        for img, target in tqdm(dataset, desc="Validating"):
+            prediction = model(img.to(device)[None])[0]
+            for k in prediction.keys():
+                prediction[k] = prediction[k].to(torch.device("cpu"))
+            metric.update([prediction], [target])
+    computed_metrics = metric.compute()
+    return computed_metrics["map_50"]
+
+
+def main():
+    # Download and prepare the COCO128 dataset
+    dataset_path = download_dataset()
+    weights_name = "SSD300_VGG16_Weights.DEFAULT"
+    transform = torchvision.models.get_weight(weights_name).transforms()
+    dataset = COCO128Dataset(dataset_path, lambda img, target: (transform(img), target))
+
+    # Get the pretrained ssd300_vgg16 model from torchvision.models
+    model = torchvision.models.get_model("ssd300_vgg16", weights=weights_name)
+    device = torch.device('cuda') if torch.cuda.is_available() else torch.cpu
+    model.to(device)
+    model.eval()
+
+    # Patch some methods with no_nncf_trace context in order for the model to be properly traced by NNCF
+    patch_with_no_nncf_trace(GeneralizedRCNNTransform.normalize)
+    patch_with_no_nncf_trace(SSD.postprocess_detections)
+
+    # Quantize model
+    calibration_dataset = nncf.Dataset(dataset, lambda data: data[0][None])
```

Comment:
It's looks good! Maybe is better to explicitly implement transform function for teaching:)

Reply:
Done